prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -0,0 +1,92 @@
+package gov.cdc.prime.router.tokens
+
+import com.google.common.net.HttpHeaders
+import com.microsoft.azure.functions.ExecutionContext
+import com.microsoft.azure.functions.HttpRequestMessage
+import com.microsoft.azure.functions.HttpResponseMessage
+import gov.cdc.prime.router.Organization
+import gov.cdc.prime.router.azure.*
+import org.apache.logging.log4j.kotlin.Logging
+
+class OktaAuthentication(private val minimumLevel: PrincipalLevel = PrincipalLevel.USER): Logging {
+    private val missingAuthorizationHeader = HttpUtilities.errorJson("Missing Authorization Header")
+    private val invalidClaim = HttpUtilities.errorJson("Invalid Authorization Header")
+
+    fun getAccessToken(request: HttpRequestMessage<String?>): String? {
+        // RFC6750 defines the access token
+        val authorization = request.headers[HttpHeaders.AUTHORIZATION.lowercase()] ?: return null
+        return authorization.substringAfter("Bearer ", "")
+    }
+
+    companion object: Logging {
+        private var httpRequestMessage: HttpRequestMessage<String?>? = null
+
+        fun setRequest(request: HttpRequestMessage<String?>) {
+            httpRequestMessage = request
+        }
+
+        val claimVerifier: AuthenticationVerifier by lazy {
```

Comment:
I think this will not do what you want -- since this is set only once, and since settings setup is usually the very first thing that runs, then all local calls to the endpoint after that will go to TestAuthenticationVerifier, even calls not from put-local-settings.py.

If you want to use a header parameter to "turn off" authentication locally, I suggest changing this to a function call, that gets the Verifier on-the-fly, on every endpoint call.   

Also, this ability to "turn off' auth occasionally seems like a general thing, not just for Settings Setup.  Maybe just give it a more general name?

I'm also worried about someone inadvertently messing up this code somehow, and this capability leaking into Prod.

Reply:
@mauricereeves tells me that we plan to soon require even local dev usage to go thru Otka.  This would imply that we just get rid of the TestAuthenticationVerifier.